website/docs/r/repository.html.markdown

```diff
@@ -124,7 +124,9 @@ initial repository creation and create the target branch inside of the repositor
 
 The `pages` block supports the following:
 
-* `source` - (Required) The source branch and directory for the rendered Pages site. See [GitHub Pages Source](#github-pages-source) below for details.
+* `source` - (Optional) The source branch and directory for the rendered Pages site. See [GitHub Pages Source](#github-pages-source) below for details.
+
+* `build_type` - (Optional) The type of GitHub Pages site to build. Can be `legacy` or `workflow`. If you use `legacy` as build type you need to set the option `source`.
```

Comment:
What happens if both source and build_type are present, which one takes precedence? What happens if legacy is set for build_type but source isn't set?

Reply:
**Allowed Options**

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
    build_type = "legacy"
  }
}
```

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
  }
}
```

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    build_type = "workflow"
  }
}
```

**Problematic / wrong Configuration**

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
    build_type = "workflow"
  }
}
```